### PR TITLE
bpo-42226: Simplify imghdr.what, and docstring, and fix logic bug

### PR DESCRIPTION
**Pull request by jph00**
imghdr.what was missing a docstring; added.

imghdr.what does not set f if h is passed, but still passed f to tests functions. None of the tests functions use it - they would not be able to anyway since it is not always set.. The unused f param has been removed from all tests, and is no longer passed along.

imghdr.what had a complex highly nest structure with multiple return paths which made the logic hard to follow. The are now on nested statements, and only one return path, and the function is now more concise and clear.

https://bugs.python.org/issue42226

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@jph00

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by jph00** [comment]
Can someone add the `skip news` label? This doesn't add or change functionality, so I assume it doesn't need a news file?

**Comment by jph00** [comment]
If this is accepted, my next hope is to move `maybe_open` into the `io` module or somewhere similar, and make it a documented, tested, and supported part of the stdlib. I'd be interested to hear if this might be of interest, and if so, what a suitable name and module location might be, and what the next steps should be (I assume it would be to create an issue with a proposal?)

**Comment by jph00** [comment]
Oh I just realized that the odd API of `tests` is actually part of the documented API, so I guess it can't easily be changed. I'll just close this because supporting that API is pretty much tied to the current implementation.
